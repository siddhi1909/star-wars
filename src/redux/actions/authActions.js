export const LOGIN_USER_SUCCESS = "LOGIN_USER_SUCCESS";
export const LOGIN_USER_FAILURE = "LOGIN_USER_FAILURE";

export const loginUserSucess = value => ({
    type: LOGIN_USER_SUCCESS,
    payload: value || "",
});

export const loginUserFailure = value => ({
    type: LOGIN_USER_FAILURE,
    payload: value,
});