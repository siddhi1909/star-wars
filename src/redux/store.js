// // IMPORT EXTERNAL LIBRARIES/MODULES
import {createStore} from 'redux';
// IMPORT ROOT REDUCER
import rootReducer from './reducer/rootReducer';

const initial_state = {};
// create the store
const store = createStore(rootReducer, initial_state);

export default store;
