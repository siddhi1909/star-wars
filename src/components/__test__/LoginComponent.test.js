// IMPORT EXTERNAL LIBRARIES/MODULES
import React from 'react';
import {Provider} from "react-redux";
import renderer from 'react-test-renderer';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
// IMPORT API & ROUTE ACTIONS
import store from "../../redux/store";
import ConnectedLoginComponent, {LoginComponent} from '../LoginComponent';
import {loginUserSucess} from '../../redux/actions/authActions';

configure({adapter: new Adapter()});

describe('>>>LOGIN --- REACT-REDUX (Mount + wrapping in < Provider >', () => {
    const initialState = {currentUser:'Luke Stywalker'}
    const mockStore = configureStore()
    let component, stores;

    beforeEach(()=>{
        stores = mockStore(initialState)
        component = mount(<Provider store={store}><ConnectedLoginComponent/></Provider>);
    })

    it('+++ render < LoginComponent /> component correctly', () => {
        const renderedValue =  renderer.create(<Provider store={store}><ConnectedLoginComponent/></Provider>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

    it('+++ render the connected(SMART) component', () => {
        expect(component.find(ConnectedLoginComponent).length).toEqual(1)
    });

    it('+++ check action on dispatching ', () => {
        let action;        
        stores.dispatch(loginUserSucess('Luke Stywalker'))
        action = stores.getActions()
        expect(action[0].type).toBe('LOGIN_USER_SUCCESS')
    });

    it('should have a btn component', () => {
        //There should be only one button
        expect(component.find('form')).toHaveLength(1);

        //Button should have matching text
        expect(component.find('button').text()).toEqual('Sign in');
    });

    it('should have input for email and password', () => {
        //Email and password input field should be present
        expect(component.find('input#username')).toHaveLength(1);
        expect(component.find('input#password')).toHaveLength(1);
    });
    
    it('should have an empty email and password state var', () => {
        //Optionally test to check if password and email are empty strings on setup
        expect(component.find(LoginComponent).state('username')).toEqual('');
        expect(component.find(LoginComponent).state('password')).toEqual('');
    });

});