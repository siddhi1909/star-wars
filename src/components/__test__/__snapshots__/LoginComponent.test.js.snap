// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>LOGIN --- REACT-REDUX (Mount + wrapping in < Provider > +++ render < LoginComponent /> component correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="logo"
      src="star-wars-logo.png"
    />
  </div>
  <div
    className="container"
  >
    <form
      className="form-box"
      id="login"
      onSubmit={[Function]}
    >
      
      <div
        className="form-group"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <div
          className="invalid-feedback"
        >
          Please enter a valid username.
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <div
          className="invalid-feedback"
        >
          Please enter your password to continue.
        </div>
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Sign in
      </button>
    </form>
  </div>
</div>
`;
